sema/commons/j2/j2_functions.py

```
@@ -46,43 +52,78 @@ def xsd_format_boolean(content, quote, suffix):
+        if str(content) != str(asint):
+            raise ValueError(
+                "int format does not round-trip "
+                f"[ {str(content)} <> {str(asint)} ]"
+            )
         content = asint
     # serialize to string again
     return xsd_value(str(content), quote, "xsd:integer")
 
 
-def xsd_format_double(content, quote, suffix):
+def xsd_format_double(content: any, quote: str, *_) -> str:
```

Ruff (0.8.2)

Missing type annotation for `*_`

(ANN002)